packages/react/src/hooks/useAutoId/useAutoId.spec.ts

```diff
@@ -0,0 +1,25 @@
+import { describe, expect, it } from '@jest/globals';
+import { renderHook } from '@testing-library/react-hooks';
+import { useAutoId } from './useAutoId';
+
+describe('useAutoId', () => {
+  it('should generate with same unique prefix', () => {
+    const { result: result1 } = renderHook(() => useAutoId('castor_test'));
```

Comment:
I don't see why this dependency is needed here, `useAutoId` and hooks in general are just pure functions, should be easy to mock and unit test.

```ts
jest.mock('react', () => ({ useMemo: (fn) => fn(), }));

it('should be easy to mock', () => {
  const autoId = useAutoId('castor_test');

  expect(autoId).toBe('castor_test_1');
});
```